patches/tModLoader/Terraria/Main.cs.patch

```diff
@@ -2648,6 +2648,193 @@
++					hovering = true;
++					player[myPlayer].mouseInterface = true;
++					if (mouseLeft && mouseLeftRelease) {
++						SoundEngine.PlaySound(12);
++						mouseLeftRelease = false;
++						if (InfoDisplayLoader.ActivePages() != InfoDisplayLoader.InfoDisplayPage + 1)
++							InfoDisplayLoader.InfoDisplayPage += 1;
++						else
++							InfoDisplayLoader.InfoDisplayPage = 0;
++					}
++
++					if (!mouseText) {
++						text = (InfoDisplayLoader.ActivePages() != InfoDisplayLoader.InfoDisplayPage + 1) ? "Next Page" : "To First Page";
++						mouseText = true;
++					}
++				}
++				spriteBatch.Draw(buttonTexture, buttonPosition, new Rectangle(0, 0, buttonTexture.Width, buttonTexture.Height), Color.White, 0f, default, 1f, SpriteEffects.None, 0f);
++				if (hovering)
++					spriteBatch.Draw(TextureAssets.InfoIcon[13].Value, buttonPosition - Vector2.One * 2f, null, OurFavoriteColor, 0f, default, 1f, SpriteEffects.None, 0f);
++
++				hovering = false;
++				GetInfoAccIconPosition(0, startX, out X, out Y);
++				buttonPosition = new Vector2(X, Y + 20);
++				if ((float)mouseX >= buttonPosition.X && (float)mouseY >= buttonPosition.Y && (float)mouseX <= buttonPosition.X + (float)buttonTexture.Width && (float)mouseY <= buttonPosition.Y + (float)buttonTexture.Height && !PlayerInput.IgnoreMouseInterface) {
++					hovering = true;
++					player[myPlayer].mouseInterface = true;
++					if (mouseLeft && mouseLeftRelease) {
++						SoundEngine.PlaySound(12);
++						mouseLeftRelease = false;
++						if (InfoDisplayLoader.InfoDisplayPage != 0)
++							InfoDisplayLoader.InfoDisplayPage -= 1;
++						else
++							InfoDisplayLoader.InfoDisplayPage = InfoDisplayLoader.ActivePages() - 1;
++					}
++
++					if (!mouseText) {
++						text = (InfoDisplayLoader.InfoDisplayPage != 0) ? "Previous Page" : "To Last Page";
++						mouseText = true;
++					}
++				}
++				spriteBatch.Draw(buttonTexture, buttonPosition, new Rectangle(0, 0, buttonTexture.Width, buttonTexture.Height), Color.White, 0f, default, 1f, SpriteEffects.FlipHorizontally, 0f);
++				if (hovering)
++					spriteBatch.Draw(TextureAssets.InfoIcon[13].Value, buttonPosition - Vector2.One * 2f, null, OurFavoriteColor, 0f, default, 1f, SpriteEffects.None, 0f);
++			}
++
+-			for (int i = 0; i < 12; i++) {
++			for (int i = startingDisplay; i < endingDisplay; i++) {
+ 				string text2 = "";
+ 				string text3 = "";
++
++				/*
+ 				if (player[myPlayer].accWatch > 0 && !flag && (!player[myPlayer].hideInfo[0] || playerInventory)) {
+ 					num = 0;
+ 					text3 = Lang.inter[95].Value;
+@@ -34941,33 +_,39 @@
+ 					text2 = text7 + " " + text6;
+ 					flag2 = true;
+ 				}
++				*/
++
++				InfoDisplay info = InfoDisplayLoader.InfoDisplays[i];
++				if (!info.Active() || (player[myPlayer].HideInfo[info.Type] && !playerInventory))
++					continue;
++
++				text2 = info.DisplayValue();
++				text3 = info.DisplayName;
+ 
```

Comment:
should be able to reduce patch size by doing some things like `int i = num3;` and `int num = info.Type`